flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamConfig.java

```diff
@@ -107,13 +107,20 @@
     private static final String TIME_CHARACTERISTIC = "timechar";
 
     private static final String MANAGED_MEMORY_FRACTION_PREFIX = "managedMemFraction.";
+
     private static final ConfigOption<Boolean> STATE_BACKEND_USE_MANAGED_MEMORY =
             ConfigOptions.key("statebackend.useManagedMemory")
                     .booleanType()
                     .noDefaultValue()
                     .withDescription(
                             "If state backend is specified, whether it uses managed memory.");
 
+    private static final ConfigOption<Boolean> SUPPORTS_UNALIGNED_INPUT =
+            ConfigOptions.key("unaligned_input")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription("Flag whether all input exchanges support unaligned input.");
+
```

Comment:
Why do we have this flag both in the `StreamConfig` and `StreamEdge`? Why not just one place?

Reply:
`StreamConfig` is used on task level to say that no input supports unaligned checkpoint. It's used while setting up the controller for `CheckpointedInputGate`. So this is input side.
On output side, you could have one downstream task that supports UC and one that doesn't. So for each `RecordWriterOutput`, we need to know that individually, hence `StreamEdge`.

On input side, we could also try to use `StreamEdge` but we would need to pass the information from `StreamTask` to `InputProcessorUtil`, which seemed uglier to me than adding that information also to `StreamConfig`.

Reply:
Changed it by using a forced aligned checkpoint. So not valid anymore.